### ISSUE TITLE
Error not detected by rust-analyzer on VsCode

### ISSUE THREAD

**Issue by rxdiscovery** [rust-lang/rust-analyzer#16990]
Hello,

I've noticed that rust-analyzer doesn't detect this syntax error on VSCode.

```rust
fn main() {
    println!("Hello, world!");
}

struct ClassA {}

impl ClassA {
    pub fn test(e: u32) {
        println!("{}", e);
    }

    println!("{}", sssa); // <---- prohibited here !!!!!!!!!!!!!!!!!
}
```

but if you compile the project or run the "cargo check" command, the error is detected.

Normally, we should get an error that tells us : "the usage of `println!` is likely invalid in impl item context"

to test the error, simply create a new empty project and open it on VSCode, then copy this code into main.rs

**Comment by Veykril** [comment]
We do diagnose this, but it currently requires `"rust-analyzer.diagnostics.experimental.enable": true,` (not sure why this one is experimental)
![image](https://github.com/rust-lang/rust-analyzer/assets/3757771/536077a9-fc60-474f-8ed7-af43dd3ca409)

**Comment by rxdiscovery** [comment]
@Veykril  Thank you very much for your answer
